Phonog-patch-1 updated to current master

lacklustr opened the pull request:
I've merged rogerclarkmelbourne/Arduino_STM32::master with Phonog-patch-1.

I also fixed a [few redundant files and other issues.](https://github.com/rogerclarkmelbourne/Arduino_STM32/pull/355)
Problem is that I just released the pixie dust from my MCP2551 adapters (won't chooch).  So i can't test this until replacements arrives. 

So, don't merge this just yet.

Phonog committed:
Add files via upload

Phonog committed:
Add files via upload

Phonog committed:
Add files via upload

Phonog committed:
Add files via upload

megadrifter committed:
added the simplified send example

also added testing sketch for GM LAN.

megadrifter committed:
Add 33 and 95 kbps speeds

megadrifter committed:
RPM reading example

Example says when you need to shift up or shift down

megadrifter committed:
add CAN_GPIO_PA11_PA12

megadrifter committed:
add A11-A12 REMAP

megadrifter committed:
add real timer setting as mcs/72

megadrifter committed:
add setCycles to use instead of setPeriod

because I neet faster interrupt than 1 microsecond.

megadrifter committed:
rename to setCycles function

megadrifter committed:
=Add dual CAN example

stevstrong committed:
Merge remote-tracking branch 'refs/remotes/rogerclarkmelbourne/master'

victorpv committed:
Merge pull request #11 from rogerclarkmelbourne/master

Sync to Roger.

victorpv committed:
Update usb_serial.h

victorpv committed:
Update usb_serial.h

victorpv committed:
Added readBytes function to usbSerial

Add Arduino USB implementation of Arduino Stream class readBytes, which reads bytes in blocks with a timeout.
Increases the speed from 300KB/s to 500KB/s over the Stream class readBytes function.

rogerclarkmelbourne committed:
Merge pull request #308 from victorpv/master

Added USBSerial implementation of readBytes.

victorpv committed:
Merge pull request #12 from rogerclarkmelbourne/master

Merge pull request #308 from victorpv/master

victorpv committed:
Add files via upload

victorpv committed:
Changes tabs for spaces.

rogerclarkmelbourne committed:
Added setClock() to Wire and Hardware - only 100000Hz and 400000Hz are supported

rogerclarkmelbourne committed:
Merge branch 'work' of https://github.com/stevstrong/Arduino_STM32 into merge_stevstrong

rogerclarkmelbourne committed:
Merge branch 'merge_stevstrong'

rogerclarkmelbourne committed:
Merge pull request #278 from syfre/master

Enhancements and fixes to use ADC in continuous mode triggered by an internal timer

rogerclarkmelbourne committed:
Merge pull request #312 from victorpv/SPI-Callbacks

Added Spi callbacks from @victorpv

Testato committed:
Merge branch 'master' into patch-4

rogerclarkmelbourne committed:
Merge pull request #303 from Testato/patch-4

Update F4 boards.txt to address memory size display issue

rogerclarkmelbourne committed:
Merge pull request #236 from Testato/patch-1

Update F1 boards.txt to address memory size display issues

mtiutiu committed:
added teensy like ide compiler opts menus

mtiutiu committed:
added missing menu opts for F3/F4 boards

stevstrong committed:
Merge remote-tracking branch 'refs/remotes/rogerclarkmelbourne/master'

rogerclarkmelbourne committed:
Merge pull request #220 from stevstrong/master

Updates for F4 from @stevstrong

hemalchevli committed:
Fixed headings

rogerclarkmelbourne committed:
Merge pull request #314 from hemalchevli/master

Fixed readme.md headings

stevstrong committed:
Merge remote-tracking branch 'refs/remotes/rogerclarkmelbourne/master'

stevstrong committed:
Merge remote-tracking branch 'refs/remotes/rogerclarkmelbourne/master'

stevstrong committed:
reworked RTC - no need for extra time library

http://www.stm32duino.com/viewtopic.php?f=39&t=775&p=31218#p31218

rogerclarkmelbourne committed:
Merge pull request #315 from stevstrong/RTC_F4

Reworked RTC for F4

rogerclarkmelbourne committed:
Change Wire endTransmission to have bool instead of unit8 argument - to match with the Arduino API

stevstrong committed:
Merge remote-tracking branch 'refs/remotes/rogerclarkmelbourne/master' into RTC_F4

stevstrong committed:
Merge remote-tracking branch 'refs/remotes/rogerclarkmelbourne/master'

stevstrong committed:
Merge remote-tracking branch 'refs/remotes/rogerclarkmelbourne/master'

stevstrong committed:
Merge branch 'refs/heads/RTC_F4'

stevstrong committed:
SPI clean up + add 16 bit access functions in 8 bit mode

rogerclarkmelbourne committed:
Merge branch 'stevstrong-SPI_write16'

rogerclarkmelbourne committed:
Merge branch 'mtiutiu/compiler_opts_menus' of https://github.com/mtiutiu/Arduino_STM32 into mtiutiu-mtiutiu/compiler_opts_menus

rogerclarkmelbourne committed:
Changed order of optimisation options, so that current settings (-Os) are the default, and reverted F4 and F3 boards.txt to current master to remove these changes from those boards, as they will need to be updated in the same way I did for the F1, and I don't have time to do it now

rogerclarkmelbourne committed:
Merge branch 'mtiutiu-mtiutiu/compiler_opts_menus'

rogerclarkmelbourne committed:
revert changes to STM32F3/platform.txt and STM32F4/platform.txt as they should not have been changed in the previous commit

rogerclarkmelbourne committed:
Fixed typo in optimisation menu

rogerclarkmelbourne committed:
Fixed yet another of my typos in the optimisation menu

rogerclarkmelbourne committed:
Added CPU Speed menu to all boards except Nucleo RB, with additional entry for 128Mhz Turbo - NO USB. Note only tested on the Blue Pill

rogerclarkmelbourne committed:
Fixed typo in F1 boards.txt for STM32F103CB RAM size

rogerclarkmelbourne committed:
Changed text on 128Mhz CPU speed option - to add warning that there is No USB and also Manual reset needed to upload

rogerclarkmelbourne committed:
Removed Debug with LTO optimisation option, to fix issue #320

rogerclarkmelbourne committed:
Added makeWord macro and functions to fix #316

mtiutiu committed:
added menu optimizations support for F3/F4 boards

rogerclarkmelbourne committed:
Merge pull request #321 from mtiutiu/mtiutiu/compiler_opts_menus

Menu optimizations support for F3/F4 boards

rogerclarkmelbourne committed:
Changed default Wire (I2C) to use hardware I2C rather than bit banged software. The existing software Wire implementation has been retained as SoftWare.  Examples have also been updated to include Wire, HardWire and Softwire.

mtiutiu committed:
linux fix for gd32 variant in boards.txt

rogerclarkmelbourne committed:
Merge pull request #322 from mtiutiu/mtiutiu/gd32_variant_linux_fix

linux fix for gd32 variant in boards.txt

rogerclarkmelbourne committed:
Fix typo in dma.h (thanks to @stevestrong)

victorpv committed:
Correcting FreeRTOS900 examples

The examples were still including the 821 version of FreeRTOS, so changing them to include the right one.

victorpv committed:
Correcting FreeRTOS900 example.

rogerclarkmelbourne committed:
Merge pull request #325 from victorpv/patch-1

Correcting FreeRTOS900 examples to include right version

rogerclarkmelbourne committed:
Added pushColors() to ILI9341 library to better support the OV7670 camera

rogerclarkmelbourne committed:
Default Wire to 100khz

victorpv committed:
Adding new.cpp as in F1 core.

This is to avoid libstdc++ being pulled in when new() is used.
Already included in the F1 in 2016.
Discussed on this thread:
http://www.stm32duino.com/viewtopic.php?f=39&t=2460&p=33163

rogerclarkmelbourne committed:
Merge pull request #327 from victorpv/patch-2

Adding new.cpp to the F4 as in F1 core.

rogerclarkmelbourne committed:
Wire lib change buffer size definition to BUFFER_LENGTH for better AVR compatibility

rogerclarkmelbourne committed:
Removed support for GD32 - by removing entry from boards.txt

rogerclarkmelbourne committed:
Update Arduino IDE version in readme

stevstrong committed:
Update RTClock.cpp

bugfix setAlarmATime - corrected day variable

csnol committed:
Rewrite the RTClock library.

csnol committed:
Create a RTC example for new library.

csnol committed:
Add STM32F103-RTC-Serial-Set.jpg

csnol committed:
Update RTClock.cpp

csnol committed:
Update RTClock.cpp

janchrillesen committed:
add JLink upload support for OSX

rogerclarkmelbourne committed:
Merge pull request #337 from janchrillesen/master

Add JLink upload support for OSX

rogerclarkmelbourne committed:
Add partial support for a DFU bootloader for the F4

rogerclarkmelbourne committed:
Merge branch 'master' of https://github.com/rogerclarkmelbourne/Arduino_STM32

rogerclarkmelbourne committed:
Fix for map() function - supplied by @Pito

victorpv committed:
Correction to upload issue in Sloeber

As discussed in http://stm32duino.com/viewtopic.php?f=41&t=2535
Changes to the Generic Boards options so the maple dfu option is the default one and Sloeber uploads correctly.

rogerclarkmelbourne committed:
Updated readme

victorpv committed:
F1 SDIO initial version

victorpv committed:
F1 SDIO

victorpv committed:
SDIO

victorpv committed:
F1 SDIO

victorpv committed:
F1 SDIO

victorpv committed:
Correct issue with port speed not changing.

The current implementation will not change speed if Wire.setClock is called after begin() since the flags are only applied when the port is enabled from being disabled.
Corrected that by adding 2 lines to disable the port, and then enable it again with the new settings.
Tested and confirmed the new speed is applied.

victorpv committed:
Update Wire.cpp

Adds a check to verify if the port had been already enabled, and only in that case disables/re-enables the port. This also solves the issue of setClock being called after begin(), but doesn't not enable the port if begin() had not been called yet.

csnol committed:
add attachAlarmInterrupt() detachAlarmInterrupt()

csnol committed:
add attachAlarmInterrupt and detachAlarmInterrupt

csnol committed:
added TimeZone(time_t xxx, timezone);

Usage:  
time_t localtime;
localtime = TimeZone(UnixTime, 8);   // Beijing timezone = 8

csnol committed:
change TZ from unit_8 to int, negatives timezone

usage: localtime = TimeZone(UnixTime, -7);   // CA USA = -7

csnol committed:
Tested on BluePill  for new function of RTC

megadrifter committed:
CAN filter corrected by  tjb12345

http://www.stm32duino.com/viewtopic.php?p=34279#p34295

csnol committed:
Add keywords for new functions

csnol committed:
Update example for Chinese friends.  timezone and filename error

csnol committed:
Add removeAlarm function

rogerclarkmelbourne committed:
Merge pull request #340 from victorpv/patch-4

Correct issue with I2C port speed not changing after begin()

rogerclarkmelbourne committed:
Merge branch 'patch-2' of https://github.com/victorpv/Arduino_STM32 into victorpv-patch-2

rogerclarkmelbourne committed:
Merge branch 'victorpv-patch-2'

rogerclarkmelbourne committed:
Merge pull request #338 from victorpv/patch-2

Correction to upload issue in Sloeber

rogerclarkmelbourne committed:
Merge branch 'master' of https://github.com/rogerclarkmelbourne/Arduino_STM32

stevstrong committed:
Create hooks.c

similar to F1 - it was missing

rogerclarkmelbourne committed:
Removed unused cygwin dll's from tools/win

stevstrong committed:
Update graphicstest.ino

- use Adafruit_GFX_AS lib instead of Adafruit_GFX

stevstrong committed:
Update onoffbutton.ino

use Adafruit_GFX_AS lib instead of Adafruit_GFX

stevstrong committed:
Update onoffbutton_breakout.ino

- use Adafruit_GFX_AS lib instead of Adafruit_GFX

stevstrong committed:
Update spitftbitmap.ino

- use Adafruit_GFX_AS lib instead of Adafruit_GFX

stevstrong committed:
Update touchpaint.ino

- use Adafruit_GFX_AS lib instead of Adafruit_GFX

greiman committed:
SPI add const qualifier

Add const qualifier to SPI member function arguments for write and
transmit buffers.

sweetlilmre committed:
Removed non standard functionality from Print class for BYTE "base" as this is available via e.g. Serial.write and conflicts with the FatFS library and potentially anything else defining BYTE
Fixed sketches using this behaviour

victorpv committed:
Update to util_adc.c file to pass pointer correctly.

This was wrong and causing warnings during compilation.

stevstrong committed:
Update touchpaint.ino

changed to correct header file

stevstrong committed:
Update boards.cpp

- remove repeated call to setup serial USB

rogerclarkmelbourne committed:
Merge pull request #354 from stevstrong/patch-9

Update boards.cpp [F4] - remove repeated call to setup serial USB

rogerclarkmelbourne committed:
Merge pull request #353 from victorpv/patch-2

Update to util_adc.c file to pass pointer correctly.

rogerclarkmelbourne committed:
Merge pull request #352 from sweetlilmre/master

Removed deprecated functionality from Print class for BYTE "base"

rogerclarkmelbourne committed:
Merge branch 'SPI_const_qualifier' of https://github.com/greiman/Arduino_STM32 into greiman-SPI_const_qualifier

rogerclarkmelbourne committed:
Merge branch 'greiman-SPI_const_qualifier'

rogerclarkmelbourne committed:
Merge pull request #350 from stevstrong/patch-8

Update touchpaint.ino [Adafruit_ILI9341_STM]

rogerclarkmelbourne committed:
Merge pull request #349 from stevstrong/patch-7

Update spitftbitmap.ino [Adafruit_ILI9341_STM]

rogerclarkmelbourne committed:
Merge pull request #348 from stevstrong/patch-6

Update onoffbutton_breakout.ino [Adafruit_ILI9341_STM]

rogerclarkmelbourne committed:
Merge pull request #347 from stevstrong/patch-5

Update onoffbutton.ino [Adafruit_ILI9341_STM]

rogerclarkmelbourne committed:
Merge pull request #346 from stevstrong/patch-4

Update graphicstest.ino [Adafruit_ILI9341_STM]

rogerclarkmelbourne committed:
Merge pull request #342 from stevstrong/patch-3

Create hooks.c [F4]

rogerclarkmelbourne committed:
Merge pull request #339 from victorpv/patch-3

F1 SDIO first version

rogerclarkmelbourne committed:
Merge pull request #333 from stevstrong/patch-2

Update RTClock.cpp [F4]

rogerclarkmelbourne committed:
Fixed issue in OLED_I2C lib caused by change to HardWire as default I2C

rogerclarkmelbourne committed:
Update ILI9341 STM lib with version from SteveStrong, and also added ReadPixelsRGB24 to steve's version

rogerclarkmelbourne committed:
Merge pull request #334 from csnol/master

Update F103 RTClock library for RTC sketch.

megadrifter committed:
latest files from Phonog

megadrifter committed:
latest files from Phonog

csnol committed:
Half-hour TimeZone supported

csnol committed:
updated

Squonk42 committed:
Added missing SDIO pins for F103V and F103Z

rogerclarkmelbourne committed:
Merge pull request #360 from Squonk42/master

Added missing SDIO pins for F103V and F103Z

csnol committed:
Update TimeZone function

Add  TimeZone(UnixTime, HourOffset, MinuteOffset)

csnol committed:
Update

csnol committed:
update

rogerclarkmelbourne committed:
Merge pull request #362 from csnol/master

Add TimeZone(UnixTime, HourOffset, MinuteOffset)

rogerclarkmelbourne committed:
Added DMA enable and disable functions to hardware timers, thanks to @stevstrong

rogerclarkmelbourne committed:
Merge branch 'master' of https://github.com/rogerclarkmelbourne/Arduino_STM32

stevstrong committed:
Update tone.cpp

- make tone() blocking for the input duration time

rogerclarkmelbourne committed:
Merge pull request #369 from stevstrong/patch-2

Update tone.cpp to fix issue if no duration was passed etc

rogerclarkmelbourne committed:
Correct argument type bug in utoa (itoa.c itoa.h) See PR #271

LightningStalker committed:
Update BlinkNcount.ino

Uses `#define` for LED pin instead of 5 literals

LightningStalker committed:
Update BlinkNcount.c

Changed pin to PC13

rogerclarkmelbourne committed:
Changed serial upload speed to 115200 as this speed is compatible with most USB to Serial adaptors

LightningStalker committed:
Merge pull request #1 from rogerclarkmelbourne/master

Changed serial upload speed to 115200 as this speed is compatible wit…

LightningStalker committed:
Update BlinkNcount.ino to use LED_BUILTIN

rogerclarkmelbourne committed:
Fixed exmple for HardTimerAsEncoder being in wrong location

rogerclarkmelbourne committed:
Fixed missing definition of digitalPinToInterrupt, by adding macro which simply uses the pin number - as it doesnt need to be mapped

rogerclarkmelbourne committed:
Fixed #273

stevstrong committed:
Update timer.h

- clear the IRQ flag before enabling the IRQ

stevstrong committed:
Update boards.txt

remove duplicated lines

albertoa committed:
Solve race condition with arduino IDE monitoring serial port

I encountered a race condition upon which, even though the udev rules were working and uploads succeed, the serial monitor can not re-establish communications.

If the serial monitor is left open when doing an upload the error shown is:

[...]
Copying data from PC to DFU device
Starting download: [##################################################] finished!
state(8) = dfuMANIFEST-WAIT-RESET, status(0) = No error condition is present
Done!
Resetting USB to switch back to runtime mode
Waiting for /dev/ttyACM0 serial...Done
processing.app.SerialException: Error opening serial port '/dev/ttyACM0'. Try consulting the documentation at http://playground.arduino.cc/Linux/All#Permission
	at processing.app.Serial.<init>(Serial.java:145)
	at processing.app.Serial.<init>(Serial.java:82)
	at processing.app.SerialMonitor$4.<init>(SerialMonitor.java:101)
	at processing.app.SerialMonitor.open(SerialMonitor.java:101)
	at processing.app.AbstractMonitor.resume(AbstractMonitor.java:104)
	at processing.app.Editor.resumeOrCloseSerialMonitor(Editor.java:2218)
	at processing.app.Editor.access$2200(Editor.java:79)
	at processing.app.Editor$DefaultExportHandler.run(Editor.java:2196)
	at java.lang.Thread.run(Thread.java:748)
Error opening serial port '/dev/ttyACM0'. Try consulting the documentation at http://playground.arduino.cc/Linux/All#Permission

I traced it to the device being opened prior to the permissions/ownership changes.

When testing via -c the upload script waits for the character device creation, but at that point the udev rules have not changed the permissions. By testing the readability of the device via the -r test, we wait until the new permissions have been applied.

I imagine this is specific to the Linux kernel and the speed of the machine being used. In my case:
arduino-1.8.5
Linux 4.4.0-98-generic #121-Ubuntu SMP
but the change should work on any system with proper udev rules / permissions.

rogerclarkmelbourne committed:
Merge pull request #375 from stevstrong/patch-3

Update timer.h to fix issue with false interrupt when first enabling it

rogerclarkmelbourne committed:
Merge pull request #376 from stevstrong/patch-4

Update boards.txt to fix issue with duplicate lines in Maple mini for clock speed selection

rogerclarkmelbourne committed:
Merge pull request #377 from albertoa/patch-1

Solve race condition, on Linux, where the Arduino IDE has the serial port (monitor) open when upload is selected

rogerclarkmelbourne committed:
Merge pull request #371 from LightningStalker/master

Update BlinkNcount.ino to use LED_BUILTIN rather than a pin number (which is specific to one board).
Note. LED_BUILTIN needs to be defined for some variants where its missing

lacklustrlabs committed:
Suppressing 'warning: missing initializer for member ... [-Wmissing-field-initializers]'

lacklustrlabs committed:
Suppressing 'warning: unused parameter ... [-Wunused-parameter]'

lacklustrlabs committed:
Fixed 'warning: multi-line comment [-Wcomment]'

lacklustrlabs committed:
Fixed 'warning: missing initializer for member ... [-Wmissing-field-initializers]'

lacklustrlabs committed:
Fixed a comment

lacklustrlabs committed:
Rollback on __attribute__((unused)) for pulseIn()

lacklustrlabs committed:
Rollback WireBase.begin()

lacklustrlabs committed:
Modified pulseIn() to use the state parameter

lacklustrlabs committed:
Caching IDR pointer

topin89 committed:
Update HardwareTimer.h documentation.

"attachInterrupt" and "attachInterrupt" functions can attach ISRs to channel 0,
which is overflow interrupt (update interrupt in RMs).

stevstrong committed:
Update TimerInterrupts.ino

- replace deprecated function

victorpv committed:
Correct a bug in FreeRTOS 

As listed in https://www.freertos.org/a00104.html#mailing_list:
"The definition of the StaticTask_t structure is incorrect if INCLUDE_xTaskAbortDelay is set to 1, or the MPU is used. The corrected structure definition can be obtained from here."
https://sourceforge.net/p/freertos/code/HEAD/tree/trunk/FreeRTOS/Source/include/FreeRTOS.h#l978

rogerclarkmelbourne committed:
Merge pull request #391 from stevstrong/patch-5

Update TimerInterrupts.ino

lacklustrlabs committed:
According to STM32F103 RM0008: TIMx_CCMR1::CC2S=01 -> channel is configured as input, IC2 is mapped on TI2.

lacklustrlabs committed:
Adhere to the format of #323

lacklustrlabs committed:
Rollback to master

lacklustrlabs committed:
Removed pinMode from PIN_MAP

rogerclarkmelbourne committed:
Merge pull request #392 from victorpv/patch-2

Correct a bug in FreeRTOS

rogerclarkmelbourne committed:
Merge pull request #386 from topin89/patch-1

Update HardwareTimer.h documentation/comments

rogerclarkmelbourne committed:
Merge branch 'Wmissing-field-initializers' of https://github.com/lacklustrlabs/Arduino_STM32 into lacklustrlabs-Wmissing-field-initializers

rogerclarkmelbourne committed:
Merge branch 'lacklustrlabs-Wmissing-field-initializers'

rogerclarkmelbourne committed:
Merge branch 'new_pulseIn' of https://github.com/lacklustrlabs/Arduino_STM32 into lacklustrlabs-new_pulseIn

rogerclarkmelbourne committed:
Merge branch 'lacklustrlabs-new_pulseIn'

rogerclarkmelbourne committed:
Merge branch 'Wunused-parameter' of https://github.com/lacklustrlabs/Arduino_STM32 into lacklustrlabs-Wunused-parameter

rogerclarkmelbourne committed:
Merge branch 'lacklustrlabs-Wunused-parameter'

rogerclarkmelbourne committed:
Modification to allow SSD1306 example to compile, following changes to Wire to use HWire by default. Note this code is only known to compile , it may not work with the SSD1306 display

aster94 committed:
Add files via upload

aster94 committed:
Add files via upload

aster94 committed:
Add files via upload

aster94 committed:
Add files via upload

aster94 committed:
Add files via upload

aster94 committed:
Delete WireBase.h

aster94 committed:
Delete WireBase.cpp

aster94 committed:
Add files via upload

aster94 committed:
Delete i2c_scanner_hardwire.ino

aster94 committed:
Update i2c_scanner_wire.ino

rogerclarkmelbourne committed:
Merge branch 'aster94-master'

rogerclarkmelbourne committed:
Hopefully fixed problem with tone() blocking when it should be non-blocking

stevstrong committed:
Create new file "atomic.h"

as discussed here: https://github.com/rogerclarkmelbourne/Arduino_STM32/issues/403, taken from http://www.stm32duino.com/viewtopic.php?f=3&t=258&start=10

rogerclarkmelbourne committed:
Merge branch 'stevstrong-patch-9'

rogerclarkmelbourne committed:
Moved atomic.h for better AVR compatibility

tfry-git committed:
Fix compilation with -std=gnu++11

See https://github.com/rogerclarkmelbourne/Arduino_STM32/issues/402

lacklustrlabs committed:
Added Wire_slave.h (no actual i2c slave implementation yet)

lacklustrlabs committed:
Removed some accidental adds

lacklustrlabs committed:
Made sure Wire_slave.h is included first

lacklustrlabs committed:
spellcheccck

lacklustrlabs committed:
keywords.txt Copied from ArduinoCore-sam (Due)

lacklustrlabs committed:
Steals any reference to Wire.h and redirects it to Wire_slave.h.
This way we don't need to alter third party libraries.
Note that Wire_slave.h must be included **before** the real Wire.h for this to work.

lacklustrlabs committed:
Added SoftWire.h

lacklustrlabs committed:
Updated library.properties

lacklustrlabs committed:
Temporarily removed not yet functioning examples

victorpv committed:
Adding PB2 as in the Generic R and Z versions.

victorpv committed:
Add PB2 to enums as in the R and Z Generic boards

Can be used for output

victorpv committed:
Add comment to PB2 pin about boot1 function.

lacklustrlabs committed:
Fixed a master/slave mode mixup.
Turns out that (sr2&I2C_SR2_MSL)!=I2C_SR2_MSL indicates slave mode after master has requested, and received, data.

"MSL - Cleared by hardware after detecting a Stop condition on the bus or a loss of arbitration
(ARLO=1), or by hardware when PE=0." -RM0008

lacklustrlabs committed:
Merged with Wire implementation from stm32duino/Arduino_Core_STM32

lacklustrlabs committed:
Saved some 50K flash memory by modifying the singleton implementation

lacklustrlabs committed:
Changed type of transmitting to bool (from uint8_t)

stevstrong committed:
Update wirish_debug.cpp

reserve SWD lines only for debug

rogerclarkmelbourne committed:
Merge pull request #419 from stevstrong/patch-10

Update wirish_debug.cpp - to ONLY reserve SWD pa13 & pa14 SWD, rather than all JTAG debug pins, which also include PB3 and PB4

lacklustrlabs committed:
Fixed minor example issues

lacklustrlabs committed:
Full method signatures in header file.
<pet peeve>
I don't understand why anyone wants to remove useful API information from the header file.
CPP files describes implementation details, header files describes the API.
</pet peeve>

lacklustrlabs committed:
Removed a redundant buffer constraint

stevstrong committed:
Update HW_STM32.h

remove static directive

rogerclarkmelbourne committed:
Merge pull request #421 from stevstrong/patch-10

Fix bug in OLED_I2C library

Commit:
option XTAL16M for boards with 16MHz-crystal

tested on STM32F103VE only but surely can be used on other STM32F103xC, STM32F103xD, STM32F103xE with the same clock PLL structure

Commit:
corrected RAM-length = 48kByte for STM32F103VC

stevstrong committed:
Update boards.txt

added error LED pin for generic F103C

rogerclarkmelbourne committed:
Merge pull request #407 from tfry-git/work/constexpr2

Fix compilation problem in STM32ADC.h when using in -std=gnu++11

rogerclarkmelbourne committed:
Merge pull request #427 from stevstrong/patch-10

Added error LED pin for generic F103C into boards.txt

rogerclarkmelbourne committed:
Merge pull request #425 from BlackBrix-Editor/changed-RAM-length-in-stm32f103vc.ld

Corrected RAM-length = 48kByte in linker file for STM32F103VC

rogerclarkmelbourne committed:
Merge pull request #424 from BlackBrix-Editor/16MHz-crystal-option

Added support inside the Core, for boards which have a 16Mhz crystal using '#define XTAL16M'. Note, no variants or boards.txt changes have been added to use this feature,

rogerclarkmelbourne committed:
Merge pull request #411 from victorpv/patch-2

Adding PB2 as in the Generic R and Z versions.

rogerclarkmelbourne committed:
Manually committed changes from https://github.com/rogerclarkmelbourne/Arduino_STM32/pull/401, as that PR included changes to permissions on unrelated files in the tools

lacklustrlabs committed:
Added platform.local.txt and boards.local.txt to .gitignore

rogerclarkmelbourne committed:
Merge pull request #430 from lacklustrlabs/Wcomment

Fixed typo causing  'warning: multi-line comment [-Wcomment]'

rogerclarkmelbourne committed:
Merge pull request #429 from lacklustrlabs/gitignore_X.local.txt

Added platform.local.txt and boards.local.txt to .gitignore

rogerclarkmelbourne committed:
Manually committed RTClock files, due to problems with PR https://github.com/rogerclarkmelbourne/Arduino_STM32/pull/428

rogerclarkmelbourne committed:
Merge branch 'TIMER_CCMR1_CC2S_INPUT_TI1' of https://github.com/lacklustrlabs/Arduino_STM32 into lacklustrlabs-TIMER_CCMR1_CC2S_INPUT_TI1

rogerclarkmelbourne committed:
Merge branch 'lacklustrlabs-TIMER_CCMR1_CC2S_INPUT_TI1'

rogerclarkmelbourne committed:
Merge branch 'WireSlave' of https://github.com/lacklustrlabs/Arduino_STM32 into lacklustrlabs-WireSlave

rogerclarkmelbourne committed:
Merge branch 'lacklustrlabs-WireSlave'

lacklustrlabs committed:
Post rebase fixup

lacklustrlabs commented:
Ok, i finally (!) got new hardware and I will start testing ~~this~~ next week.

This is a big rebase, if you want to compare this vs the current Arduino_STM32:Master you can look here: https://github.com/lacklustrlabs/Arduino_STM32/tree/CAN_library

(don't merge yet)

rogerclarkmelbourne committed:
Update BOARD_NR_GPIO_PINS for STM32F103V

This fixes problem caused by https://github.com/rogerclarkmelbourne/Arduino_STM32/commit/951d0acf178b28d52919fb777874d90603c88883#diff-3df511e94e3cba57d187bb1322b9f6c6

megadrifter commented:
Hello. Just to ask, am I to do anyshing with this?
